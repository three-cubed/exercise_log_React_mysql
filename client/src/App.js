import React, { useState, useEffect, useCallback } from 'react';
import './stylesheet.css';
import ExerciseEventList from './components/ExerciseEventList';
import ExerciseAdder from './components/ExerciseAdder';
import TitleBlock from './components/TitleBlock';

function App() {
    const [exerciseEvents, setExerciseEvents] = useState([]);
    const [showForm, setShowForm] = useState(false);

    const getExerciseList = useCallback(() => {
        console.log('getExerciseList() is running');
        fetch('/getExercise')
          .then(res => res.json())
          .then(setExerciseEvents);
    });

    useEffect(() => {
        getExerciseList();
    }, []); 

    const executeOnClick = () => {
        setShowForm(!showForm);
    }

    const restateAfterDelete = () => {
        getExerciseList();
    }

    const addExerciseEvent = ( addExerciseEventArguments ) => {
        fetch('/exercisePost', {
            method: 'post',
            headers: { 'Content-Type': 'application/json' },
            body: JSON.stringify(addExerciseEventArguments)
        }).then(() => { 
            console.log('message from App.js: posted, should now getExerciseList()');
            getExerciseList();
        });
    }
    
    return (
        <div className="App">
            <TitleBlock 
                onClick={executeOnClick}   
                btnText={showForm ? 'Close new exercise event' : 'Add a new exercise event'}  
                btnColour={showForm ? ' #ffcc99' : 'rgba(103, 189, 103, 0.74)'}
            />
            {
            showForm    
            &&
            <ExerciseAdder addExerciseEvent={addExerciseEvent} />
            }
            {showForm && <br />}
            <br />
            <ExerciseEventList 
                exercises={exerciseEvents} 
                restateAfterDelete={restateAfterDelete}
            />
        </div>
    );
}

export default App;
