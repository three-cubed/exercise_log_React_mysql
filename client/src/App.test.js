import React from 'react';
import App from './App';
import { cleanup, fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';

afterEach(cleanup);

describe('App component', () => {
    test('renders', () => {
        render(<App />);
    })
    test('matches snapshot', () => {
        const tree = renderer.create(<App />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('App uses ExerciseAdder form', () => {
    test('does not show form (ByTestId) before top btn click', () => {
        const newApp = render(<App />);
        expect(newApp.queryByTestId('exerciseAdderForm')).toBeNull();
    });
    test('shows form (ByTestId) after top btn click', () => {
        const newApp = render(<App />);
        fireEvent.click(newApp.getByText(/add a new exercise/i));
        expect(newApp.getByTestId('exerciseAdderForm')).not.toBeNull();
    });
    test('does not show submit btn (ByText) before top btn click', () => {
        const newApp = render(<App />);
        expect(newApp.queryByText(/Save new exercise event/i)).toBeNull();
    });
    test('shows submit btn (ByText) after top btn click', () => {
        const newApp = render(<App />);
        fireEvent.click(newApp.getByText(/Add a new exercise/i));
        expect(newApp.getByText(/Save new exercise event/i)).not.toBeNull();
    });
});
