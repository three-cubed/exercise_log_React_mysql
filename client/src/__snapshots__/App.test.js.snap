// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="App"
>
  <div
    className="titleBlock"
  >
    <h2>
      The Exercise Log
    </h2>
    <div
      className="centred"
    >
      <button
        className="titleBlockBtn"
        onClick={[Function]}
        style={
          Object {
            "backgroundColor": "rgba(103, 189, 103, 0.74)",
          }
        }
        type="button"
      >
        Add a new exercise event
      </button>
    </div>
  </div>
  <br />
</div>
`;
